charts/egeria-platform/templates/statefulset.yaml

```diff
@@ -0,0 +1,119 @@
+  labels:
+    {{- include "egeria-platform.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  serviceName: {{ include "egeria-platform.fullname" . }}
+  selector:
+    matchLabels:
+      {{- include "egeria-platform.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "egeria-platform.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "egeria-platform.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: egeria
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          envFrom:
+            - configMapRef:
+                name: {{ include "egeria-platform.fullname" . }}-config
+            {{ if .Values.secretName }}
+            - secretRef:
+                name: {{ .Values.secretName }}
+            {{- end }}
+          ports:
+            - containerPort: {{ .Values.service.platformPort }}
+              name: platform
+          {{ if .Values.egeria.debug }}
+            - containerPort: {{ .Values.service.debugPort }}
+              name: debug
+          {{ end }}
+          # No other checks until this passes
+          startupProbe:
+            exec:
+              command:
+                - stat
+                - /tmp/done.txt
+            initialDelaySeconds: 10
+            periodSeconds: 5
+            timeoutSeconds: 1
+            failureThreshold: 25
+          # Is pod ready to service network requests - it will pause (for replicas, others will take the load)
+          readinessProbe:
+            httpGet:
+              path: /open-metadata/platform-services/users/{{ .Values.egeria.user }}/server-platform/origin
+              port: {{ .Values.service.platformPort }}
+              scheme: HTTPS
+            periodSeconds: 10
+            failureThreshold: 25
+          # Is pod doing useful work - if not we will restart it
+          livenessProbe:
+            httpGet:
+              path: /open-metadata/platform-services/users/{{ .Values.egeria.user }}/server-platform/origin
+              port: {{ .Values.service.platformPort }}
+              scheme: HTTPS
+            periodSeconds: 10
+            failureThreshold: 25
+          command: ["/home/jboss/scripts/run.sh"]
+          resources:
+            {{- toYaml .Values.resources | nindent 12 }}
+          {{ if .Values.egeria.persistence }}
```

Comment:
has the chart been tested with and without persistence? Do the scripts cause any issues ie with changing of metadata ids, or do they only run once (not tested myself)

Reply:
The chart has not been tested without persistence.
The scripts only run if no configuration documents are found (see `scripts/run.sh`), so they should only run once.

Reply:
Perhaps add a comment to the values.yaml to mention this?
But feel free to close this thread - all the charts are examples anyway